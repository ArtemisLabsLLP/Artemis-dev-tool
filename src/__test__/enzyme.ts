// import React from 'react';
// import { configure, shallow } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import toJson from 'enzyme-to-json'


// configure({ adapter: new Adapter() });

// describe('React unit tests', () => {
//     describe('LabeledText', () => {
//         let wrapper;
//         const props = {
//             label: '',
//             text: '.'
//         };

//         beforeAll(() => {
//             wrapper = shallow(<LabeledText {...props} />);
//         })

//         it('Snapshot testing labeled test', () => {
//             expect(toJson(wrapper)).toMatchSnapshot();
//         })

//         it('Renders a <p> tag with the label in bold', () => {
//             expect(wrapper.type()).toEqual('p');
//             expect(wrapper.text()).toEqual('Mega: Markets');
//             expect(wrapper.find('strong').text()).toMatch('Mega');
//           });
//         });
      
//         describe("MarketDisplay", () => {
//           // TODO: Test the following:
//           // 1. A MarketDisplay should display all of its text props inside a
//           // LabeledText component
//           // 2. It should also contain a div with two buttons
//           // 3. The functions passed down should be invoked on click
//           // 4. MarketDisplay should render a div with a class of `innerbox`, and the
//           // interior div wrapping the two buttons should have a class of `flex`
      
//           let wrapper;
//           let marketBox;
      
//           const props = {
//             index: 0,
//             location: "los angeles",
//             cards: 10,
//             percentage: 100,
//             addCard: jest.fn(),
//             deleteCard: jest.fn()
//           };
      
//           beforeAll(() => {
//             wrapper = shallow(<MarketDisplay {...props} />);
//             marketBox = wrapper.find(".marketBox");
//           });
      
//           it('Snapshot testing', () => {
//             expect(toJson(wrapper)).toMatchSnapshot();
//           })
      
//           it("MarketDisplay should display all of its text props inside a LabeledText component", () => {
//             const propValues = Object.values(props);
//             wrapper.find(LabeledText).forEach(node => {
//               expect(propValues.includes(node.prop("text"))).toBe(true);
//             });
//           });
      
//           it("Contains a div with two buttons", () => {
//             // console.log(wrapper.find("div").find("button").length) //you can log to the console the length 
      
//             // const buttonList = wrapper.find("div").find("button"); //can save wrapper to a cariable and then 
//             // expect(buttonList).toHaveLength(2);
//             // expect(buttonList.length).toBe(2);
      
//             // expect(wrapper.find("div").find("button").length).toBe(2); //can chain .find 's
//             expect(wrapper.children("div").children("button").length).toBe(2); //can chain .children 's
//           });
      
//           it("Passed down functions should be invoked onClick", () => {
//             marketBox.find("button").forEach(node => {
//               node.simulate("click");
//             });
//             // expect(props.cards).toBe(1);
//             expect(props.addCard).toHaveBeenCalled();
//             expect(props.deleteCard).toHaveBeenCalled();
//           });
      
//           it('Should render a div with class of "marketBox" with two buttons containing the class "flex"', () => {
//             expect(marketBox.at(0).hasClass("marketBox")).toBe(true);
//             const flex = marketBox.find(".flex");
//             expect(flex.hasClass("flex")).toBe(true);
//           });
//         });
      
      
//          // TODO: Test the following:
//           //   1. A MarketsDisplay should have an h4 element to display the 'Markets'
//           //   title
//           //   2. A single MarketDisplay is rendered for each market in the
//           //   marketList prop
//           //   3. The percentage prop should be a string calculated to two decimals.
//           //   Test for zero, a whole number, and a fractional value. (Right now this
//           //   is implemented incorrectly, so follow TDD here)
      
//         describe("MarketsDisplay", () => {
//           let wrapper;
//           let markets;
//           const props = {
//             totalCards: 500,
//             marketList: [
//               { location: "Los Angeles", cards: 200 },
//               { location: "Glendale", cards: 237 },
//               { location: "Las Vegas", cards: 63 },
//               { location: "New York City", cards: 0 }
//             ]
//           };
//           beforeAll(() => {
//             wrapper = shallow(<MarketsDisplay {...props} />);
//             markets = wrapper.find(MarketDisplay);
//           });
      
//           it('Snapshot testing', () => {
//             expect(toJson(wrapper)).toMatchSnapshot();
//           })
          
//           it("Should have an h4 element to display the 'Markets' title", () => {
//             const title = wrapper.find("h4");
//             expect(title.text()).toBe("Markets");
//           });
      
//           it("A single MarketDisplay is rendered for each market in the marketList prop", () => {
//             expect(markets.length).toBe(4);
//           });
      
//           it("Percentage prop should be a string calculated to 2 decimals", () => {
//             //test for zero
//             expect(markets.at(3).prop("percentage")).toBe(0);
//             //whole num
//             expect(markets.at(0).prop("percentage")).toBe("40.00");
//             //fractional value
//             expect(markets.at(1).prop("percentage")).toBe("47.40");
//           });
//         });
//       });
      