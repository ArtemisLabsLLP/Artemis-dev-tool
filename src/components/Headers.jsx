import React, { useState, useEffect } from 'react';

const Headers = (props) => {
  const [status, updateStatus] = useState('query');

  useEffect(() => {
    if(status === 'schema') updateStatus('query');
    else if(status === 'cache') updateStatus('response');
    else if(status === 'query') updateStatus('schema');
    else if(status === 'response') updateStatus('cache');
  },[status])

  // function schemaToggle() {
    
  // }

  // function cacheToggle() {

  // }
  
  return (
    <div>
      <div onClick={()=> schemaToggle()}></div>
      <div onClick={()=> cacheToggle()}></div>
    </div>
  )
}

export default Headers;