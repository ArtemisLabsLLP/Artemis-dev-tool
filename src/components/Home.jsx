import React from 'react';

const Home = props => {
    return(
        <div id='home-component'>
            <h1>ARTEMIS</h1>
            <p>GraphQL query not has been detected yet.</p>
        </div>
    )
}

export default Home;