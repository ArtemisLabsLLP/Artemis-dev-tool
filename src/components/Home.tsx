import React from 'react';


const Home :React.FC=()=> {

    return (
        <div id='home-component'>
            <h1>ARTEMIS</h1>
            <p>GraphQL query has not been detected yet.</p>
            <div className="loader">Loading...</div>
        </div>
    )
}

export default Home;