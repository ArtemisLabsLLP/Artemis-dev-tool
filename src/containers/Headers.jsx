import React from 'react';
import Dropdown from '../components/DropdownMenu.jsx';
import { Container, Button, Alert } from 'react-bootstrap';
import { CSSTransition } from 'react-transition-group';

const Headers = (props) => {
  // const [showButton, setShowButton] = useState(true);
  // const [showMessage, setShowMessage] = useState(false);

  return (
    <React.Fragment>
      <div className="hOne">
        <h1>ARTEMIS</h1>
        <img></img>
      </div>
      <div className="header-container">
        <Dropdown/>
        {/* <Container> */}
        {/* <button className="header-item item2" onClick={()=> {
          props.schemaToggle();
          setShowMessage(true);
          }
          }> */}
        <button className="header-item item2" onClick={()=> props.schemaToggle()}>
          SCHEMA
        </button>
        <button className="header-item item3" onClick={()=> props.cacheToggle()}>
          CACHE
        </button>
        {/* </Container> */}
      </div>
    </React.Fragment>
  )
}

export default Headers;