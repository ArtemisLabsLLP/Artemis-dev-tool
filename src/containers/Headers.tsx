import * as React from 'react';
import Dropdown from '../components/DropdownMenu';
// import SelectCreatable, { CreatableProps } from 'react-select/lib/Creatable';
// import SelectReadOnly from 'react-select';
// // import { getStyles } from './SelectDropdownStyles';

// import 'normalize.css/normalize.css'
// import '@blueprintjs/core/dist/blueprint.css'
import Button from "../components/Button"
interface IHeaders {
  visualizerToggle: () => void,
  schemaToggle: () => void,
  cacheToggle: () => void,
}



const Headers: React.FC<IHeaders> = (props) => {

  return (
    <React.Fragment>
  
      <nav className="pt-navbar">

        <div className="pt-navbar-group pt-align-left"> 
            <div className="pt-navbar-heading"> Upload File</div>
            <input className="pt-input" placeholder="Search Query Id.." type="text" />
        </div>



      </nav>
      <div className="hOne">
        <h1>ARTEMIS</h1>
      </div>
      <div className="header-container">
        <Dropdown />
        <Button className="header-item item4" onClick={() => props.visualizerToggle()}>
          VISUALIZER
        </Button>
        <button className="header-item item2" onClick={() => props.schemaToggle()}>
          SCHEMA
        </button>
        <button className="header-item item3" onClick={() => props.cacheToggle()}>
          CACHE
        </button>
      </div>
    </React.Fragment>
  )
}

export default Headers;