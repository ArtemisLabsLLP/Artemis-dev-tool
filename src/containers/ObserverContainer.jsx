import React, { useEffect, useState } from "react";
import HistoryOfPastQueries from "../components/HistoryOfPastQueries.jsx";
import Query from "../components/Query.jsx";
import GraphQLResponse from "../components/GraphQLResponse.jsx";
import Schema from "../components/Schema.jsx";
import ApolloGraphQLCache from "../components/ApolloGraphQLCache.jsx";
import DropdownMenu from "../components/DropdownMenu.jsx";

const ObserverContainers = props => {
  // const [queries, updateQueries] = useState([]);
  // const [results, updateResults] = useState([]);
  // const [history, recordHistory] = useState([]);
  // const [historyBtn, historyBtnToggle] = useState(0);
  // const [url, updateUrl] = useState("");
  // const [cache, updateCache] = useState({});

  // const getCache = () => {
  //   msgToBackground("contentScript", "getCache", response => {
  //     console.log(response)
  //     msgToBackground("contentScript", "retrieveCache", response => { updateCache(response) });
  //   });
  // }

  // function isToggle(index) {
  //   historyBtnToggle(index);
  //   msgToBackground(
  //     "contentScript",
  //     "rerenderDOM",
  //     response => console.log(response),
  //     history[index]
  //   );
  // }

  // useEffect(() => {
  //   historyBtnToggle(queries.length - 1);
  // }, [queries]);

  // useEffect(() => {
  //   chrome.devtools.network.onRequestFinished.addListener(httpReq => {

  //     if (httpReq.request.postData) {
  //       updateUrl(httpReq.request.url);
  //       httpReq.getContent(res => {
  //         updateResults(oldResults => [...oldResults, res]);
  //       });
  //       let requestQuery;
  //       if (IsJsonString(httpReq.request.postData.text)) {
  //         requestQuery = JSON.parse(httpReq.request.postData.text).query;
  //       } else {
  //         requestQuery = httpReq.request.postData.text;
  //       }
  //       msgToBackground("contentScript", "getDOM", response =>
  //         recordHistory(oldHistory => [...oldHistory, response.msg])
  //       );
  //       updateQueries(oldQueries => [
  //         ...oldQueries,
  //         {
  //           time: httpReq.time,
  //           outgoingQueries: requestQuery
  //         }
  //       ]);
  //     }
  //   });
  // }, []);

  return (
    <React.Fragment>
      <div id="observerContainers">
        <HistoryOfPastQueries queries={queries} isToggle={isToggle} history={history}/>
        <Query queries={props.queries} historyBtn={props.historyBtn} />
        <GraphQLResponse results={props.results} historyBtn={props.historyBtn} />
        {props.schemaStatus ? <Schema historyBtn={props.historyBtn} url={props.url} queries={props.queries} /> : null}
        {props.cacheStatus ? <ApolloGraphQLCache historyBtn={props.historyBtn} url={props.url} queries={props.queries} getCache={props.getCache} cache={props.cache} /> : null}
      </div>
    </React.Fragment>
  );
};

// const msgToBackground = function (type, msg, callback, newBody) {
//   if (chrome && chrome.runtime) {
//     chrome.runtime.sendMessage(
//       {
//         type,
//         msg,
//         newBody
//       },
//       function (response) {
//         callback(response);
//       }
//     );
//   }
// };

// const IsJsonString = function (str) {
//   try {
//     JSON.parse(str);
//   } catch (e) {
//     return false;
//   }
//   return true;
// };

export default ObserverContainers;