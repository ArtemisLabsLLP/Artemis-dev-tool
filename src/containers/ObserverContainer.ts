import * as React from 'react'
import HistoryOfPastQueries from "../components/HistoryOfPastQueries.ts";
import Query from "../components/Query.ts";
import GraphQLResponse from "../components/GraphQLResponse.ts";
import Schema from "../components/Schema.ts";
import ApolloGraphQLCache from "../components/ApolloGraphQLCache.ts";
import GraphQLVisualizer from '../components/GraphQLVisualizer.ts';
// import ChartComponent from "../components/ChartComponent.jsx";

interface IProps {
    queries: object[],
    isToggle: (index: number) => void,
    history: string[],
    historyBtn: number,
    results: string[],
    visualizerStatus: boolean,
    schemaStatus: boolean,
    url: string,
    cacheStatus: boolean,
    getCache: () => void,
    cache: any,
}

const ObserverContainer: React.FC<IProps> = props => {

    return (
        <React.Fragment>
            <div id="observerContainers">
                <HistoryOfPastQueries queries={props.queries} isToggle={props.isToggle} history={props.history} />
                <Query queries={props.queries} historyBtn={props.historyBtn} />
                <GraphQLResponse results={props.results} historyBtn={props.historyBtn} />

            </div>
            {props.visualizerStatus ? <GraphQLVisualizer results={props.results} /> : null}
            {props.schemaStatus ? <Schema url={props.url} queries={props.queries} /> : null}
            {props.cacheStatus ? <ApolloGraphQLCache queries={props.queries} getCache={props.getCache} cache={props.cache} /> : null}
        </React.Fragment>
    )
}

export default ObserverContainer;
