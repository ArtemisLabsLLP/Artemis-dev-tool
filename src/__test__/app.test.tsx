import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import '../setupTests';
import  App  from '../containers/App'
import ObserverContainer from "../containers/ObserverContainer";
import Headers from '../containers/Headers';
import Home from '../components/Home';
import msgToBackground from '../Utility/msgToBackground'
import '../setupTests';


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

describe('App', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<App />);
    });
    it('Should match snapshot, render Header, Body, and Footer', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.exists(ObserverContainer)).toBe(true);
      expect(wrapper.exists(Headers)).toBe(true);
      expect(wrapper.exists(Home)).toBe(true);
      expect(wrapper.exists(msgToBackground)).toBe(false);
    });
  });