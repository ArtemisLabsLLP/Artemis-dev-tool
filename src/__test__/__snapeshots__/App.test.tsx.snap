exports[`App Should match snapshot, render Header, Body, and Footer 1`] = `
<div
  id="App"
>
  <Component
    schemaToggle={[Function]}
    cacheToggle={[Function]}
  />

  <Component
    copyToClipboard={[Function]}
    handleToggle={[Function]}
    isValidTab={true}
    recStatus="off"
  />
</div>
`;